gwpy/table/gravityspy.py

```diff
@@ -19,22 +20,83 @@
 """Extend :mod:`astropy.table` with the `GravitySpyTable`
 """
 
-import json
-import os
-from urllib.error import HTTPError
+import re
+from json import JSONDecodeError
+from pathlib import Path
 from urllib.parse import urlencode
-from urllib.request import urlopen
 
-import numpy as np
-
-from astropy.utils.data import get_readable_fileobj
+import requests
 
 from ..utils import mp as mp_utils
 from .table import EventTable
 
 __author__ = 'Scott Coughlin <scott.coughlin@ligo.org>'
 __all__ = ['GravitySpyTable']
 
+JSON_CONTENT_TYPE = "application/json"
+
+# default gravity spy remote host
+DEFAULT_HOST = "https://gravityspytools.ciera.northwestern.edu"
+
+# API paths
+SEARCH_PATH = "/search"
+SIMILARITY_SEARCH_PATH = "/similarity_search_restful_API"
+
+# ordered list of image durations supported by gravityspy.
+# NOTE: the index of each element here is the map between
+#       the 'urlX' column in the GravitySpyTable and the
+#       duration of the relevant image, for some reason.
+DURATIONS = (
+    0.5,
+    1.0,
+    2.0,
+    4.0,
+)
+
+# list of ERAs supported by similarity search API
+ERA = {
+    "ALL": (1126400000, 1584057618),
+    "O1": (1126400000, 1137250000),
+    "ER10": (1161907217, 1164499217),
+    "O2a": (1164499217, 1219276818),
+    "ER13": (1228838418, 1229176818),
+    "ER14": (1235750418, 1238112018),
+    "O3": (1238166018, 1272326418),
+    "O3a": (1238166018, 1254009618),
+    "O3b": (1256655642, 1272326418),
+}
+
+# regex to filter columns containing URLs
+URL_COLUMN = re.compile(r"\Aurl[0-9]+\Z")
+
+
+def _image_download_target(
+    row,
+    outdir,
+    training_set=False,
+    labelled_samples=False,
+):
+    """Construct where to download a GravitySpy image
+
+    This returns a `str.format`-style template that just needs
+    the duration to form a complete target path.
+    """
```

Comment:
Took me a while to figure out outdir must be a Path object and / is an overload for join. Defining the argument data types might save the next person some time